beacon-chain/cache/signed_execution_header.go

```diff
@@ -0,0 +1,87 @@
+package cache
+
+import (
+	"bytes"
+	"sync"
+
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/primitives"
+	enginev1 "github.com/prysmaticlabs/prysm/v5/proto/engine/v1"
+)
+
+var (
+	// This cache is intended for use by the sync service to store signed execution payload headers after they pass validation.
+	// The signed header from this cache could be used by the proposer when proposing the next slot.
+	cachedSignedExecutionPayloadHeader = make(map[primitives.Slot][]*enginev1.SignedExecutionPayloadHeader)
+	mu                                 sync.RWMutex
+)
+
+// SaveSignedExecutionPayloadHeader saves the signed execution payload header to the cache.
+// The cache stores headers for up to two slots. If the input slot is higher than the lowest slot
+// currently in the cache, the lowest slot is removed to make space for the new header.
+// Only the highest value header for a given parent block hash will be stored.
+// This function assumes caller already checks header's slot is current or next slot, it doesn't account for slot validation.
+func SaveSignedExecutionPayloadHeader(header *enginev1.SignedExecutionPayloadHeader) {
+	mu.Lock()
+	defer mu.Unlock()
+
+	for s := range cachedSignedExecutionPayloadHeader {
+		if s+1 < header.Message.Slot {
+			delete(cachedSignedExecutionPayloadHeader, s)
+		}
+	}
+
+	// Add or update the header in the map
+	if _, ok := cachedSignedExecutionPayloadHeader[header.Message.Slot]; !ok {
+		cachedSignedExecutionPayloadHeader[header.Message.Slot] = []*enginev1.SignedExecutionPayloadHeader{header}
+	} else {
+		found := false
+		for i, h := range cachedSignedExecutionPayloadHeader[header.Message.Slot] {
```

Comment:
Just a suggestion can't this be a separate function for better code readability something like 
``` func GetHighestBidHeader(header, cache)(header ,found)
```
If found is true set it or false append

Reply:
I thought about it and wasn't a fan. I think it's less readable and more cumbersome. First you have to think about lock, second `GetHighestBidHeader` will have to update the entry but that is what the name suggest